certification/internal/shell/imagesource_registry.go

```diff
@@ -0,0 +1,72 @@
+package shell
+
+import (
+	"bytes"
+	"strings"
+
+	cmdchain "github.com/rainu/go-command-chain"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	log "github.com/sirupsen/logrus"
+)
+
+type ImageSourceRegistryCheck struct {
+}
+
+func (p *ImageSourceRegistryCheck) Validate(bundleImage string) (bool, error) {
+
+	approvedRegistries := []string{
+		"registry.connect.dev.redhat.com",
+		"registry.connect.qa.redhat.com",
+		"registry.connect.stage.redhat.com",
+		"registry.connect.redhat.com",
+		"registry.redhat.io",
+		"registry.access.redhat.com",
+	}
+
+	output := &bytes.Buffer{}
+	inputContent := strings.NewReader(bundleImage)
+
+	err := cmdchain.Builder().
+		WithInput(inputContent).
+		Join("cut", "-d", ",", "-f1").
+		Join("cut", "-d", "/", "-f1").
+		Finalize().WithOutput(output).Run()
+	if err != nil {
+		log.Error(" Failed to execute cmdchain builder")
+		log.Debug(" failed to execute cmdchain builder", err)
+		return false, nil
+	}
+
+	userRegistry := strings.TrimRight(output.String(), "\n")
+	log.Info("Check Image registry for : ", userRegistry)
+
+	for _, val := range approvedRegistries {
```

Comment:
I believe keeping the approved registry as a map will enable you to perform a search in O(1), without having to implement a loop.

Reply:
Agreed. 
Implemented. This is note to myself that it's done and It will be in the next push.